src/account/AccountPickScreen.js

```diff
@@ -28,15 +36,21 @@ class AccountPickScreen extends PureComponent<Props> {
   static contextType = TranslationContext;
   context: GetText;
 
-  handleAccountSelect = (index: number) => {
+  handleAccountSelect = async (index: number) => {
+    const _ = this.context;
     const { accounts, dispatch } = this.props;
     const { realm, isLoggedIn } = accounts[index];
     if (isLoggedIn) {
       setTimeout(() => {
         dispatch(accountSwitch(index));
       });
     } else {
-      NavigationService.dispatch(navigateToRealmInputScreen({ realm }));
+      try {
+        const serverSettings: ApiResponseServerSettings = await api.getServerSettings(realm);
+        NavigationService.dispatch(navigateToAuth(serverSettings));
+      } catch (e) {
+        showErrorAlert(_('Failed to connect to realm: {realm}', { realm: realm.toString() }));
```

Comment:
This looks to be a new user-facing string, so it should go in the `messages_en.json` database for translation.